Extension/src/LanguageServer/client.ts

```diff
@@ -906,21 +906,13 @@ export class DefaultClient implements Client {
 
         const items: IndexableQuickPickItem[] = [];
         for (let i: number = 0; i < paths.length; i++) {
-            let option: string | undefined;
-            let isCompiler: boolean = false;
-            const slash: string = (os.platform() === 'win32') ? "\\" : "/";
+            const compiler: string = path.basename(paths[i]).toLowerCase();
+            const isCompiler: boolean = compiler !== paths[i];
```

Comment:
How come `compiler !== paths[i]` means it is a compiler?

Reply:
If it is not equal, then it is only a compiler name. For example, "cl.exe" was successfully extracted from a path. If this does it is equal, then it will be equal to the full path option (i.e one of the last three options like "Help me install a compiler"). Thus it will be categorized as false or "not a compiler"

Reply:
Oh I see, the name "compiler" made me think it had to validate it's a compiler instead of is compiler name.

Reply:
should I change it to isValidCompilerName?

Reply:
Maybe call it `compilerName` for clarity

Reply:
(not essential though)

Reply:
If you'd like to make the variable reflect what it's checking. Descriptive variables are good for readability.

Reply:
Maybe change `compiler` to `compilerName` too.